### PR TITLE
Decorator pattern: SmartTroll should be SmartHostile #264

### PR DESCRIPTION
**Pull request by tomroy**
In this request I renamed the SmartTroll-class to SmartHostile and replaced the hardcoded strings in that class to "It throws a rock at you!".
I also updated the class naming in the UML digram.

### CODE REVIEW

File: decorator/src/test/java/com/iluwatar/decorator/SmartHostileTest.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  *
  * @author Jeroen Meulemeester
  */
-public class SmartTrollTest {
+public class SmartHostileTest {
 
   @Test
   public void testSmartTroll() throws Exception {
```
Comment (iluwatar) [inline decorator/src/test/java/com/iluwatar/decorator/SmartHostileTest.java]: This method should be renamed to testSmartHostile

**Comment by iluwatar** [comment]
For some reason there are two .png images in `etc`folder. Could you remove the other one and rename the correct image to `decorator.png`?

**Comment by iluwatar** [comment]
@tomroy you have my review remarks. Once done with the changes, please comment on this thread and I will perform the merge.

**Comment by tomroy** [comment]
@iluwatar I renamed the method to `testSmartHostile` and left only the correct image named `decorator.png`.

**Comment by iluwatar** [comment]
Good job @tomroy Thank you for the contribution.

**Pull request merged**